watertap/unit_models/zero_order/tests/test_gas_sparged_membrane.py

```diff
@@ -0,0 +1,238 @@
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.solute_treated_equation[0, "cod"]) == 1e5
+
+    @pytest.mark.component
+    def test_initialization(self, model):
+        initialization_tester(model)
+
+    @pytest.mark.component
+    def test_solve(self, model):
+        results = solver.solve(model)
+
+        # Check for optimal solution
+        assert check_optimal_termination(results)
+
+    @pytest.mark.component
+    def test_solution(self, model):
+        assert (pytest.approx(979, rel=1e-5) ==
+                value(model.fs.unit.treated.flow_mass_comp[0, "H2O"]))
+        assert (pytest.approx(20.58784, rel=1e-5) ==
+                value(model.fs.unit.byproduct.flow_mass_comp[0, "H2O"]))
+        assert (pytest.approx(0, abs=1e-5) ==
+                value(model.fs.unit.treated.flow_mass_comp[0, "cod"]))
+        assert (pytest.approx(1, rel=1e-5) ==
+                value(model.fs.unit.byproduct.flow_mass_comp[0, "cod"]))
+
+        assert (pytest.approx(1.001, rel=1e-5) ==
+                value(model.fs.unit.properties_in[0].flow_vol))
+        assert (pytest.approx(0.9990001, rel=1e-5) ==
+                value(model.fs.unit.properties_in[0].conc_mass_comp["cod"]))
+        assert (pytest.approx(0.979, rel=1e-5) ==
+                value(model.fs.unit.properties_treated[0].flow_vol))
+        assert (pytest.approx(0, abs=1e-5) ==
+                value(model.fs.unit.properties_treated[0].conc_mass_comp["cod"]))
+        assert (pytest.approx(0.021588, rel=1e-5) ==
+                value(model.fs.unit.properties_byproduct[0].flow_vol))
+        assert (pytest.approx(46.32238, rel=1e-5) ==
+                value(model.fs.unit.properties_byproduct[0].conc_mass_comp["cod"]))
+        assert (pytest.approx(368.855328, abs=1e-5) ==
+                value(model.fs.unit.electricity[0]))
+        assert (pytest.approx(5*value(model.fs.unit.properties_in[0].flow_vol)*0.08235, rel=1e-5) ==
+                value(model.fs.unit.flow_mass_gas_extraction[0]))
+
+
+    @pytest.mark.component
+    def test_conservation(self, model):
+        for (t, j) in model.fs.unit.inlet.flow_mass_comp.keys():
+            if j != "H2O":
+                assert (abs(value(model.fs.unit.inlet.flow_mass_comp[t, j] -
+                                  model.fs.unit.treated.flow_mass_comp[t, j] -
+                                  model.fs.unit.byproduct.flow_mass_comp[t, j]))
+                        <= 1e-6)
+            else:
+                assert (abs(value(model.fs.unit.inlet.flow_mass_comp[t, j] -
+                                  model.fs.unit.treated.flow_mass_comp[t, j] -
+                                  model.fs.unit.byproduct.flow_mass_comp[t, j] -
+                                  model.fs.unit.flow_mass_gas_extraction[t]))
+                        <= 1e-6)
+
+    @pytest.mark.component
+    def test_report(self, model):
+        if not is_solver_from_idaes_ext:
```

Comment:
This is definitely not critical, but I just wanted to mention that you can use the `requires_idaes_solver` pytest mark for this: simply decorate the test function (you can apply multiple pytest marks), and the test will be skipped/xfailed if the solver is not one of the IDAES solvers (i.e. on macOS):

```suggestion
    @pytest.mark.requires_idaes_solver
    @pytest.mark.component
    def test_report(self, model):
        # the rest of the test, without the if clause
```